app/config/taxRatesAndBands.scala

```diff
@@ -47,15 +48,27 @@ object TaxRatesAndBands {
 
   val allRates: List[TaxRatesAndBands] = TaxRatesAndBands20152016 :: TaxRatesAndBands20162017 :: TaxRatesAndBands20172018 ::
     TaxRatesAndBands20182019 :: TaxRatesAndBands20192020 :: TaxRatesAndBands20202021 :: TaxRatesAndBands20212022 ::
-    TaxRatesAndBands20222023 :: TaxRatesAndBands20232024 :: TaxRatesAndBands20242025 :: Nil
+    TaxRatesAndBands20222023 :: TaxRatesAndBands20232024 :: TaxRatesAndBands20242025 :: TaxRatesAndBands20242025MidYearChange :: Nil
 
   val liveTaxRates: List[TaxRatesAndBands] = if (LocalDate.now.isBefore(latestTaxYearGoLiveDate)) allRates.dropRight(1) else allRates
 
-  def getRates(year: Int): TaxRatesAndBands = liveTaxRates.filter(_.taxYear == year) match {
-    case params if params.nonEmpty => params.head
+  def getRates(year: Int, disposalDate : Option[LocalDate] = None , isMidYearChangeApplicable : Boolean = false): TaxRatesAndBands = liveTaxRates.filter(_.taxYear == year) match {
+    case params if params.nonEmpty => getTaxRatesAndBandsForMidYearChange(params, disposalDate,isMidYearChangeApplicable)
     case _ => liveTaxRates.maxBy(_.taxYear)
   }
 
+  private def getTaxRatesAndBandsForMidYearChange(liveTaxRates : List[TaxRatesAndBands],disposalDate: Option[LocalDate], isMidYearChangeApplicable: Boolean): TaxRatesAndBands = {
+    if (isMidYearChangeApplicable && liveTaxRates.size > 1) {
```

Comment:
I think the if statement should be moved to the getRates function. Currently, we are always calling this function regardless if there is a mid year change

Reply:
I see your point. And, I think both could be simplified having given it another pass. Let me catchup with Shivangi and push up the change